### PR TITLE
Feature/sg 000 yolo leftovers

### PR DESCRIPTION
**Pull request by shaydeci**
- Removal of old YoloV5 objects references in YAML files.
- Removal of unused YoloV5 leftover blocks.
- Replace head method fixed for YoloX (was replacing head with Detect instead of DetectX)

### CODE REVIEW

File: src/super_gradients/recipes/training_hyperparams/coco2017_yolox_train_params.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ lr_cooldown_epochs: 15
 warmup_bias_lr: 0.0
 warmup_momentum: 0.9
 initial_lr:  0.02
-warmup_mode: yolov5_warmup
+warmup_mode: yolo_warmup
```
Comment (yurkovak) [inline src/super_gradients/recipes/training_hyperparams/coco2017_yolox_train_params.yaml:12]: You don't need this warmup mode anymore, discussed offline

File: src/super_gradients/training/models/detection_models/yolo_base.py [hunk 2]
```diff
@@ -437,27 +437,19 @@ def _check_strides_and_anchors(self):
 
         self.register_buffer('stride', m.stride)  # USED ONLY FOR CONVERSION
 
-    def _initialize_biases(self, cf=None):
+    def _initialize_biases(self):
         """initialize biases into Detect(), cf is class frequency"""
```
Comment (yurkovak) [inline src/super_gradients/training/models/detection_models/yolo_base.py:441]: *DetectX()

File: src/super_gradients/training/models/detection_models/yolo_base.py [hunk 3]
```diff
@@ -437,27 +437,19 @@ def _check_strides_and_anchors(self):
...
         detect_module = self._head._modules_list[-1]  # Detect() module
-        if isinstance(detect_module, Detect):
```
Comment (yurkovak) [inline src/super_gradients/training/models/detection_models/yolo_base.py:443]: Are you going to remove the `Detect` class definition? 
By analogy to this, please remove `if isinstance(m, Detect):` in `_check_strides_and_anchors()`. 
You can also get rid of `check_anchor_order(...)`

**Review by yurkovak — CHANGES_REQUESTED** [review]

File: tests/integration_tests/pretrained_models_test.py [hunk 4]
```diff
@@ -189,7 +189,29 @@ def setUp(self) -> None:
...
+                 "criterion_params": {
+                     "strides": [8, 16, 32],  # output strides of all yolo outputs
+                     "num_classes": len(self.transfer_detection_dataset['ssd_lite_mobilenet_v2'].classes)},
+
+                 "loss_logging_items_names": ["iou", "obj", "cls", "l1", "num_fg", "Loss"],
+
+                 "train_metrics_list": [],
+                 "valid_metrics_list": [
+                     DetectionMetrics(
+                         post_prediction_callback=SSDPostPredictCallback(),
```
Comment (yurkovak) [inline tests/integration_tests/pretrained_models_test.py:210]: Could you please explain what is added here? Is it a mistake that SSD things are used for the DetectionMetrics under yolox key or is it supposed to be like this?

File: src/super_gradients/training/models/detection_models/yolo_base.py (hunk 2 above)
Comment (shaydeci) [reply src/super_gradients/training/models/detection_models/yolo_base.py]: fixed in https://github.com/Deci-AI/super-gradients/pull/300/commits/075e5c7478335fc969e906fcd7f1a74aec904822

File: tests/integration_tests/pretrained_models_test.py (hunk 4 above)
Comment (shaydeci) [reply tests/integration_tests/pretrained_models_test.py]: Post prediction callback is a mistake, but worked since they actually do the exact same thing (:
I will to the backlog that we should drop it and just rename the yolo one.

Regarding the other stuff- I'm just using the transfer learning dataset, but since its confusing I changed it to just be the same dataset (I.e without the dictionary and all the transfer learning detection tests just use the same one)

File: src/super_gradients/training/models/detection_models/yolo_base.py [hunk 5]
```diff
@@ -413,51 +351,32 @@ def _initialize_module(self):
...
-    def update_param_groups(self, param_groups: list, lr: float, epoch: int, iter: int,
-                            training_params: HpmStruct, total_batch: int) -> list:
-
-        lr_warmup_epochs = get_param(training_params, 'lr_warmup_epochs', 0)
-        if epoch >= lr_warmup_epochs:
-            return super().update_param_groups(param_groups, lr, epoch, iter, training_params, total_batch)
-        else:
-            return param_groups
-
     def _check_strides_and_anchors(self):
```
Comment (yurkovak) [inline src/super_gradients/training/models/detection_models/yolo_base.py:354]: Sorry for the review in waves, just noticed. Do you mind renaming this to `_check_strides` please?

Comment (shaydeci) [reply src/super_gradients/training/models/detection_models/yolo_base.py]: done https://github.com/Deci-AI/super-gradients/pull/300/commits/fcdab0aac5e6a19fb95a3de35b48b88f2a96edba

**Review by yurkovak — APPROVED** [review]
LGTM
